features/step_definitions/sign_up_steps.rb

```diff
@@ -0,0 +1,44 @@
+Given("the following user exists") do |table|
+  table.hashes.each do |user|
+    User.create(user)
+  end
+end
+
+When("I fill in {string} with {string}") do |field, value|
+  fill_in field, with: value
+end
+
+When("I click button {string}") do |element|
+  click_button element
+end
+
+When("I press the {string} button") do |sign_up|
+  click_link sign_up
+end
+
+Then("I should be redirected to the {string} page") do |page_name|
+  expect(page.current_path).to eq page_path_from(page_name)
+end
+
+Then("I should be on the {string} page") do |page_name|
+  visit page_path_from(page_name)
+end
+
+Given("I am on the {string} page") do |page_name|
```

Comment:
Come to think of it even this step sounds like an assertion.. If you want this to be an action you should rather say something like

`Given I visit the "Page name" page`

then in the step definition you can perform the action of visiting that page